### ISSUE TITLE
[BUG] -Werror=catch-value made warnings as errors, then build failed.

### ISSUE THREAD

**Issue by kaigai** [rapidsai/cudf#3613]
**Describe the bug**

During the cuDF module build from the source code, it generates terrible amount of compilation errors, like:
```
/home/kaigai/anaconda3/envs/cudf_dev/include/rmm/rmm.hpp: In instantiation of 'rmmError_t rmm::alloc(T**, size_t, cudaStream_t, const char*, unsigned int) [with T = char; size_t = long unsigned int; cudaStream_t = CUstream_st*]':
/home/kaigai/repo/cudf/cpp/custrings/strings/../regex/../util.inl:96:29:   required from 'T* device_alloc(size_t, cudaStream_t) [with T = char; size_t = long unsigned int; cudaStream_t = CUstream_st*]'
/home/kaigai/repo/cudf/cpp/custrings/strings/replace_backref.cu:155:93:   required from here
/home/kaigai/anaconda3/envs/cudf_dev/include/rmm/rmm.hpp:139:3: error: catching polymorphic type 'class std::exception' by value [-Werror=catch-value=]
   } catch(std::exception e) {
```
Although I could not check all the error message (build script had been generated error message for 20 minutes...), it looks to me c/c++ compiler detected a warning related to catch exceptions, and a compiler option: `-Werror=catch-value` treated it as error.

**Steps/Code to reproduce bug**

https://github.com/rapidsai/cudf/blob/branch-0.12/CONTRIBUTING.md#script-to-build-cudf-from-source

We followed the steps described above.
`make -j` kicks a series of build chains, then raised this problem.

**Environment overview (please complete the following information)**

- OS: CentOS 8 with latest packages
- gcc-8.2.1-3.5.el8.x86_64
- cuda-nvcc-10-1-10.1.243-1.x86_64
- cmake-3.14.5-1.fc29.x86_64
    - RHEL8/CentOS8 distributed cmake-3.11, so we picked up from Fedora29
- boost-devel-1.66.0-6.el8.x86_64
- boost169-devel-1.69.0-4.el8.x86_64

**Additional comment**

I have been trying to build/install cuDF packages for more than a week, however, it does not work well.
Is it possible to distribute the RPM packages for RHEL/CentOS installation bases?

**Comment by harrism** [comment]
This looks like an RMM issue rather than a libcudf issue.

**Comment by kaigai** [comment]
OK, I try to build https://github.com/rapidsai/rmm individually.
Let me report later.

**Comment by harrism** [comment]
Even if it builds OK, it's in a header so I think we should fix it (it's a 1-character fix. :).  If you can file the same issue text above as an RMM issue that would help us assign and track it.

**Comment by harrism** [comment]
Fixed by rapidsai/rmm#215

**Comment by trevorsm7** [comment]
I think this issue is blocked on https://github.com/NVIDIA/jitify/issues/36 as well as a few changes in cudf

**Comment by trevorsm7** [comment]
I checked a similar fix into NVIDIA/jitify, but it needs to be merged into the cudf branch: https://github.com/NVIDIA/jitify/pull/37

**Comment by harrism** [comment]
Can you submit a PR to the cudf fork of Jitify to merge the latest NVIDIA/jitify?

**Comment by trevorsm7** [comment]
@trxcllnt suggested waiting on the binops PR to merge jitify (cudf is currently ~30 commits behind)
